### PR TITLE
Fix build phases not being generated in the right positions

### PR DESCRIPTION
**Pull request by pepibumur**
Fixes https://github.com/tuist/tuist/issues/505

### Short description 📝
As @parse reported, the target actions were not generated in the right position. That happens because after calling the build phases generator, we call the link phases generator which might insert phases after the post actions.

### Solution 📦
Move the calls to generate the target actions to the `TargetGenerator`.

### Implementation 👩‍💻👨‍💻
- [x] Fix the issue.
- [x] Add a unit tests.
- [x] Add an acceptance test.

### CODE REVIEW

File: features/generate.feature [hunk 1]
```diff
@@ -175,3 +175,11 @@ Scenario: The project is an iOS application with an incompatible Xcode version (
     And I have a working directory
     Then I copy the fixture ios_app_with_incompatible_xcode into the working directory
     Then tuist generates yields error "The project, which only supports the versions of Xcode 3.2.1, is not compatible with your selected version of Xcode"
+
+Scenario: The project is an iOS application with target actions
+    Given that tuist is available
+    And I have a working directory
+    Then I copy the fixture ios_app_with_actions into the working directory
+    Then tuist generates the project
+    Then the target App should have the build phase Tuist in the first position
```
Comment (pepibumur) [inline features/generate.feature]: @parse am I right defining in this test what you expect in the generated project?

Comment (parse) [reply features/generate.feature]: Correct!

Comment (pepibumur) [reply features/generate.feature]: Awesome! With this test in place we'll make sure we don't introduce regressions in the future.

File: Tests/TuistGeneratorTests/Generator/TargetGeneratorTests.swift [hunk 2]
```diff
@@ -107,6 +107,48 @@ final class TargetGeneratorTests: XCTestCase {
         ])
     }
 
+    func test_generateTarget_actions() throws {
+        /// Given
```
Comment (marciniwanicki) [inline Tests/TuistGeneratorTests/Generator/TargetGeneratorTests.swift]: I think we mostly use `//` (not `///`) for given/when/than as it's not a documentation.

**Pull request merged**